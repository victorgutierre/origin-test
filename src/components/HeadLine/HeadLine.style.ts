import styled from 'styled-components';

export const H2 = styled.h2`
  display: block;
  text-align: center;
  padding: 2rem;
  color: #1d34a1;
  font-size: 1.375rem;
  font-style: normal;
  font-weight: 400;

  @media screen and (min-width: 768px) {
    padding: 2.8rem 0 1.5rem 0;
  }
`;

H2.displayName = 'h2';
