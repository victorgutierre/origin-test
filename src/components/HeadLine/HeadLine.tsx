import React from 'react';

export const HeadLine = () => {
  return (
    <div>Headline</div>
  )
};
