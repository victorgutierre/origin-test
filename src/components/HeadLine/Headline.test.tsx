import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import { HeadLine } from './HeadLine';

describe('<Headline />', () => {
  const component = (
    <HeadLine firstSentence="Let's plan your" boldSentence="saving goal." />
  );

  it('Should render Headline component', () => {
    const wrapper = shallow(component);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Must contain h2 and strong element', () => {
    const { queryByTestId } = render(component);
    const h2 = queryByTestId('headline');
    const strong = queryByTestId('headline').querySelector('strong');
    expect(h2).toBeTruthy();
    expect(strong).toBeTruthy();
  });

  it('Must contain a full sentence', () => {
    const { queryByTestId } = render(component);
    const headline = queryByTestId('headline');
    expect(headline.textContent).toBe("Let's plan your saving goal.");
  });
});
