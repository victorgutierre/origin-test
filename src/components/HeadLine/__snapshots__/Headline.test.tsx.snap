// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline /> Should render Headline component 1`] = `
<h2
  data-testid="headline"
>
  Let's plan your
   
  <strong>
    saving goal.
  </strong>
</h2>
`;
